1309979,refresh how to pad a number with leading zero.

john-par opened the pull request:
## Summary

Content updates for [Task 1409979](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1409979?src=WorkItemMention&src-action=artifact_link).

john-par committed:
1309979, refreshed the content of the article.

rpetrusha commented on docs/standard/base-types/how-to-pad-a-number-with-leading-zeros.md:
```diff
@@ -14,18 +14,20 @@ ms.assetid: 0b2c2cb5-c580-4891-8d81-cb632f5ec384
-    -   To display the integer as a hexadecimal value, call its `ToString(String)` method and pass the string "X*n*" as the value of the `format` parameter, where *n* represents the minimum length of the string.  
+    - To display the integer as a hexadecimal value, call its `ToString(String)` method and pass the string "X*n*" as the value of the format parameter, where *n* represents the minimum length of the string.
   
      You can also use the format string in a method, such as <xref:System.String.Format%2A> or <xref:System.Console.WriteLine%2A>, that uses [composite formatting](../../../docs/standard/base-types/composite-formatting.md).  
```
```suggestion
You can also use the format string in an interpolated string in both [C#](../../csharp/language-reference/interpolated.md) and [Visual Basic](../../visual-basic/programming-guide/language-features/strings/interpolated-strings.md), or you can call a method, such as <xref:System.String.Format%2A?displayProperty=nameWithtype> or <xref:System.Console.WriteLine%2A?displayProperty=nameWithType>, that uses [composite formatting](../../../docs/standard/base-types/composite-formatting.md).  
```
String interpolation is recommended over composite formatting.

rpetrusha commented on docs/standard/base-types/how-to-pad-a-number-with-leading-zeros.md:
```diff
@@ -36,51 +38,59 @@ You can add leading zeros to an integer by using the "D" [standard numeric forma
+1. Define a custom numeric format string that uses the zero placeholder "0" to represent the minimum number of zeros.  
   
-3.  Call the number's `ToString(String)` method and pass it the custom format string. You can also use the custom format string with a method that supports composite formatting.  
+1. Call the number's `ToString(String)` method and pass it the custom format string. You can also use the custom format string with a method that supports composite formatting.  
```
```suggestion
1. Call the number's `ToString(String)` method and pass it the custom format string. You can also use the custom format string with string interpolation or with a method that supports composite formatting.  
```

rpetrusha reviewed (changes requested):
You're changes look good, @john-par. I've added two suggestions that reflect the current recommendation to use string interpolation instead of composite formatting.

Commit:
Update docs/standard/base-types/how-to-pad-a-number-with-leading-zeros.md

Co-Authored-By: john-par <v-jopar@microsoft.com>

Commit:
Update docs/standard/base-types/how-to-pad-a-number-with-leading-zeros.md

Co-Authored-By: john-par <v-jopar@microsoft.com>

john-par committed:
Update how-to-pad-a-number-with-leading-zeros.md

@rpetrusha, thank you for the suggested updates. I added `/tokens/` to the **C#** hyperlink so it pointed to the  **$ - string interpolation (C# Reference)** article.

john-par commented:
@rpetrusha, thank you for the quick feedback. I have committed your change requests with a minor update to fix a link. If you're happy with the content changes, this PR is ready for sign-off.

My colleague, William Rohm, has been assigned to [Task 1409979](https://mseng.visualstudio.com/TechnicalContent/_workitems/edit/1409979). If you'd like to see any code updates related to the task, he will take care of those. If you are happy with the content and the code in the article, please let us know and we will mark the task as **Completed**.

rpetrusha commented:
This looks good. I'll merge your PR now, @john-par.

rpetrusha commented:
In response to the question about code updates, @john-par and @WilliamAntonRohm, the calls to Console.WriteLine should be modified to use interpolated strings. ,

WilliamAntonRohm commented:
Ron -- to be clear, I'm to replace all composite formatting strings with interpolated strings?  If so, should I update all text references to "composite formatting" to instead describe "interpolated strings"?

Alternatively, for each example I could add a new set of WriteLine statements that use interpolated strings, following the two existing sets of ToString & composite-formatted statements, and make the corresponding text additions.

Please let me know your preference.
 
@john-par fyi

WilliamAntonRohm commented:
@rpetrusha -- please see above

rpetrusha commented:
Yes, @WilliamAntonRohm, composite format strings should be replaced with interpolated strings. And the text references to composite formatting should also be changed to "interpolated strings". (Thanks for reminding me -- I'd forgotten about that little detail.)
